File: 2020_leetcode/problems/__01__/python/recursion_principle.py

class RecursionPrinciple:
    def printReverse(self, str):
        def helper(index, str):
            if (index >= len(str)):
                return
            helper(index + 1, str)
            print(str[index])
        helper(0, str)

RecursionPrinciple().printReverse(list("abc"))

File: 2020_leetcode/problems/__01__/typescript/recursionPrinciple.js
var RecursionPrinciple = function () {
    var printReverse = function (str) {
        var helper = function (index, str) {
            if (index >= str.length) {
                return;
            }
            helper(index + 1, str);
            console.log(str[index]);
        };
        helper(0, str);
    };
    return {
        printReverse: printReverse
    };
};
RecursionPrinciple().printReverse("abc".split(""));
